spring-cloud-gcp-pubsub-stream-binder/src/test/java/org/springframework/cloud/gcp/stream/binder/pubsub/properties/PubSubExtendedBindingsPropertiesTests.java

```diff
@@ -0,0 +1,69 @@
+package org.springframework.cloud.gcp.stream.binder.pubsub.properties;
+
+import org.junit.Test;
+
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.cloud.gcp.autoconfigure.core.GcpContextAutoConfiguration;
+import org.springframework.cloud.gcp.autoconfigure.pubsub.GcpPubSubAutoConfiguration;
+import org.springframework.cloud.gcp.autoconfigure.pubsub.GcpPubSubEmulatorConfiguration;
+import org.springframework.cloud.gcp.stream.binder.pubsub.PubSubMessageChannelBinder;
+import org.springframework.cloud.gcp.stream.binder.pubsub.properties.PubSubExtendedBindingsPropertiesTests.PubSubBindingsTestConfiguration.CustomTestSink;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.Input;
+import org.springframework.cloud.stream.annotation.StreamListener;
+import org.springframework.cloud.stream.binder.BinderFactory;
+import org.springframework.cloud.stream.config.BindingServiceConfiguration;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.SubscribableChannel;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class PubSubExtendedBindingsPropertiesTests {
+
+	private ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+			.withPropertyValues(
+				"spring.cloud.stream.gcp.pubsub.bindings.input.consumer.auto-create-resources=true",
+					"spring.cloud.stream.gcp.pubsub.default.consumer.auto-create-resources=false")
+			.withConfiguration(AutoConfigurations.of(
+					GcpPubSubEmulatorConfiguration.class,
+					GcpContextAutoConfiguration.class,
+					GcpPubSubAutoConfiguration.class,
```

Comment:
Do you really need these first three? It will try to configure credentials and such, which I don't think is necessary for this test.

Reply:
I am investigating if these can be removed. The test seems to need the beans PubSubAdmin and pubSubChannelProvisioner. I think these are needed either by the BinderFactory or by the bindings created in PubSubBindingsTestConfiguration.

I won't be able to keep them in order for the test to run on Travis.

Reply:
You should be able to just provide mocks for those. Put the mocks into the application context.

Reply:
So it looks like there is a little bit more to it, turns out I also need the PubSubTemplate binding because it looks like my test bindings (i.e. the CustomTestSink) are being registered by the Binding service. Doing a plain mock here doesn't work since it needs to keep track of the channels that were created. I'll dig deeper and get back to you on a solution.